### ISSUE TITLE
Can't catch KeyboardInterrupt error on python3

### ISSUE THREAD

**Issue by jander** [gevent/gevent#461]
When starting a stream server on python3. gevent can't catch KeyboardInterrupt error to exit.
for example:
```
def start_server(address, handle_func):
    server = StreamServer(address, handle_func)
    print('Starting %s on %r' % ( __file__, address))
    try:
        server.serve_forever()
    except (KeyboardInterrupt, SystemExit):
        print('Exiting...')
    finally:
        server.stop()
```
`print('Exiting...')` never run on python3.

I look over `hup.py`, find line 263 is:
```
NOT_ERROR = (GreenletExit, SystemExit)
```
I think it should be:
```
NOT_ERROR = (GreenletExit, SystemExit, KeyboardInterrupt)
```

I changed that line code on my pc, and the server run correctly.

**Comment by hashstat** [comment]
#471 appears to be a duplicate of this issue and seems to have more activity.

**Comment by jamadden** [comment]
Fixed by #578.
